File: 0036-valid-sudoku/0036-valid-sudoku.ts
function isValidSudoku(board: string[][]): boolean {
    const row = Array.from({ length: 9 }, () => new Set());
    const col = Array.from({ length: 9 }, () => new Set());
    const block = Array.from({ length: 9 }, () => new Set());

    for (let i = 0; i < board.length; i++){
        for (let j = 0; j < board[0].length; j++){
            let val = board[i][j]
            if (val != '.') {
                let r = i;
                let c = j;
                let b = Math.floor(i / 3)*3 + Math.floor(j / 3)
                if (row[r].has(val) || col[c].has(val) || block[b].has(val)) {
                    return false 
                }

                row[r].add(val)
                col[c].add(val)
                block[b].add(val)
            }
                
        }   
    }

    return true
};

File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row = [set() for _ in range(len(board))]
        col = [set() for _ in range(len(board))]
        block = [set() for _ in range(len(board))]
        

        for i in range(len(board)):
            for j in range(len(board[0])):
                if board[i][j] == ".":
                    continue
                val = board[i][j]

                b = (i // 3) * 3 + (j // 3)

                if val in row[i] or val in col[j] or val in block[b]:
                    return False
                
                row[i].add(val)
                col[j].add(val)
                block[b].add(val)

        return True